### PR TITLE
gcoap: check if NDEBUG is defined

### PR DESCRIPTION
**Pull request by nmeum**
Otherwise the build breaks when `CFLAGS` contains `-DNDEBUG` but `DEVELHELP` is set since the `cf` parameter is not used in that case.

Other modules also check for NDEBUG instead of DEVELHELP e.g. `core/panic.c`: https://github.com/RIOT-OS/RIOT/blob/master/core/panic.c#L49

### CODE REVIEW

File: sys/net/application_layer/gcoap/gcoap.c [hunk 1]
```diff
@@ -813,7 +813,7 @@ uint8_t gcoap_op_state(void)
 int gcoap_get_resource_list(void *buf, size_t maxlen, uint8_t cf)
 {
     assert(cf == COAP_CT_LINK_FORMAT);
-#ifndef DEVELHELP
```
Comment (smlng) [inline sys/net/application_layer/gcoap/gcoap.c]: IMHO both are superfluous, the `cf` parameter isn't used other than in the `assert`, I would remove the preprocessor regardless of `DEVELHELP` or `NDEBUG` and leave the `(void)cf;` standalone.

Make the code more clean and result will also be the same (code size wise and so on).

TL;DR: better remove `#if[n]def`.

Comment (nmeum) [reply sys/net/application_layer/gcoap/gcoap.c]: Well…I actually disagree since the `#ifdef` makes it more clear that the variable is **only** used in the assert-statement but since you seem to see this differently I changed it anyhow.

**Comment by smlng** [comment]
My point was in favour of readability and I also think is makes no difference binary size or execution time wise. But as we two disagree here, lets get a third (or forth) opinion here.

Summoning @kaspar030  and @gebart 😉 Any comments on this? That is, preference for remove or leave but change to `NDEBUG`?

**Comment by kaspar030** [comment]
+1

**Comment by smlng** [comment]
so removal it is.
